File: python/examples/basic/cluster/cl_closed.py
"""
Closed cluster: 8 jobs cycle between Think delay and 3 PS servers.
"""

import numpy as np

from line_solver import (GlobalConstants, MVA, Network, RoutingStrategy,
                         SchedStrategy, VerboseLevel)


if __name__ == "__main__":
    GlobalConstants.set_verbose(VerboseLevel.STD)

    N = np.array([8])           # population
    Z = np.array([1.0])          # think time
    D = np.full((3, 1), 1.0)    # mean service time at each server
    strategies = [SchedStrategy.PS] * 3

    model = Network.cluster_closed(N, Z, D, strategies, dispatching=RoutingStrategy.RAND)
    print(MVA(model).get_avg_table())

File: matlab/examples/basic/cluster/cl_closed.m
% SF_CLOSED  Closed cluster: 8 jobs cycling between Think and 3 PS servers.

clear all

N = 8;
Z = 1.0;
D = ones(3, 1);
strategy = {SchedStrategy.PS, SchedStrategy.PS, SchedStrategy.PS};

model = Network.clusterClosed(N, Z, D, strategy, ones(3, 1), RoutingStrategy.RAND);

solver = MVA(model);
disp(solver.getAvgTable());
